tests/unit/amo/components/TestMoreAddonsByAuthorsCard.js

```diff
@@ -0,0 +1,456 @@
+      store: dispatchClientMetadata().store,
+      ...customProps,
+    };
+
+    return shallowUntilTarget(
+      <MoreAddonsByAuthorsCard {...props} />,
+      MoreAddonsByAuthorsCardBase
+    );
+  }
+
+  it('should render a card', () => {
+    const { store } = dispatchClientMetadata();
+    const authorNames = ['test', 'test2', 'test3'];
+    store.dispatch(loadAddonsByAuthors({
+      addons: Object.values(fakeAddons()),
+      authorNames,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames,
+      store,
+    });
+
+    expect(root).toHaveClassName('MoreAddonsByAuthorsCard');
+  });
+
+  it('should render a className', () => {
+    const { store } = dispatchClientMetadata();
+    const authorNames = ['test', 'test2', 'test3'];
+    store.dispatch(loadAddonsByAuthors({
+      addons: Object.values(fakeAddons()),
+      authorNames,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames,
+      className: 'foo',
+      store,
+    });
+
+    expect(root).toHaveClassName('foo');
+  });
+
+  it('should render nothing if there are no add-ons', () => {
+    const { store } = dispatchClientMetadata();
+    const authorNames = ['test', 'test2', 'test3'];
+    store.dispatch(loadAddonsByAuthors({
+      addons: [],
+      authorNames,
+    }));
+
+    const root = render({ authorNames, store });
+
+    expect(root).not.toHaveClassName('MoreAddonsByAuthorsCard');
+    expect(root.html()).toBeNull();
+  });
+
+  it('should render nothing if add-ons are null', () => {
+    const root = render({
+      addons: null,
+      authorNames: ['test2'],
+    });
+
+    expect(root).not.toHaveClassName('MoreAddonsByAuthorsCard');
+    expect(root.html()).toBeNull();
+  });
+
+  it('should render a card with loading state if loading', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    store.dispatch(fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandlerId: errorHandler.id,
+    }));
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    expect(root).toHaveProp('loading', true);
+    expect(root).toHaveClassName('MoreAddonsByAuthorsCard');
+  });
+
+  it('should dispatch a fetch action if no add-ons found', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandlerId: errorHandler.id,
+    }));
+  });
+
+  it('should dispatch a fetch action if props are updated', () => {
```

Comment:
There's no test for just changing the `addonType` prop and not also the `authorNames` prop (although there is one for just changing the `authorNames` prop and not the `addonType` prop), so I think that should be added. I'm not sure if it's better to make this test case longer, or have separate cases for `addonType` and `authorNames` (as you have for `forAddonSlug` below).